docs/guides/testing-dbus.rst

```diff
@@ -0,0 +1,290 @@
+Testing D-Bus Services
+======================
+Class :class:`sssd_test_framework.utils.sbus.SBUSUtils` provides an API to
+operate on D-Bus services. This class has two methods:
+
+* :meth:`sssd_test_framework.utils.sbus.SBUSUtils.getInfoPipe()`
+* :meth:`sssd_test_framework.utils.sbus.SBUSUtils.getDestination()`
+
+It is important to notice that you don't need to instantiate an object of class
+:class:`sssd_test_framework.utils.sbus.SBUSUtils` because the ``Client`` role already
+provides one when you invoke the
+:meth:`sssd_test_framework.roles.client.Client.sbus()` method.
+
+:meth:`sssd_test_framework.utils.sbus.SBUSUtils.getInfoPipe()` returns a preinstantiated
+object of class :class:`sssd_test_framework.utils.sbus.DBUSDestination`, which gives you
+direct access to the `Infopipe` destination; while the method
+:meth:`sssd_test_framework.utils.sbus.SBUSUtils.getDestination()` allows you to
+create a new instance of :class:`sssd_test_framework.utils.sbus.DBUSDestination`
+for which you can specify the actual `destination` (a server) and the ``bus`` to
+use to contact that destination.
+
+.. code-block:: python
+    :caption: Getting a destination
+
+    ifp_dest = client.sbus().getInfoPipe()
+    monitor_dest = client.sbus().getDestination(dest="sssd.monitor", bus=DBUSKnownBus.MONITOR)
+
+
+The bus to provide to :meth:`sssd_test_framework.utils.sbus.SBUSUtils.getDestination()`
+must be an object of class :class:`sssd_test_framework.utils.sbus.DBUSKnownBus` which
+provides some predefined buses:
+
+* :attr:`sssd_test_framework.utils.sbus.DBUSKnownBus.SYSTEM`
+* :attr:`sssd_test_framework.utils.sbus.DBUSKnownBus.SESSION`
+* :attr:`sssd_test_framework.utils.sbus.DBUSKnownBus.MONITOR`
+
+(If no bus is provided, the system bus is used.)
+
+But it also provides the class method
+:meth:`sssd_test_framework.utils.sbus.DBUSKnownBus.forDomain()` which generates
+the bus object for the domain whose name was passed as parameter.
+
+.. code-block:: python
+    :caption: Accessing the destination ``sssd.nss`` on domain ``MyDomain``
+
+    bus = DBUSKnownBus.forDomain("MyDomain")
+    dest = client.sbus().getDestination(dest="sssd.nss", bus=bus)
+
+With the destination, it is possible to get the different objects it provides. Each object
+has its own ``object path``. These objects are obtained by calling the destination's
+:meth:`sssd_test_framework.utils.sbus.DBUSDestination.getObject()` method. The object path
+must be passed as argument. A list conataining all the available object paths can be retrieved
+with the method :meth:`sssd_test_framework.utils.sbus.DBUSDestination.getObjectPaths()`.
+
+.. note::
+    Passing an ``object path`` that doesn't exist will not generate an error nor
+    raise an exception. It will, instead, create a valid object with no attributes.
+    This is a strange behavior of ``Introspection``.
```

Comment:
What is introspection? Is this some python feature?

Reply:
Long story short, introspection is a mechanism provided by D-Bus for objects to describe themselves. The description includes which methods and properties it provides, their signatures, and its children.

Reply:
Thanks!